src/connector/Connector.ts

```diff
@@ -0,0 +1,145 @@
+import { uid } from 'uid';
+import { ICustomEventMessageResponse } from './ICustomEventMessageResponse';
+import { ICustomEventMessageRequest } from './ICustomEventMessageRequest';
+import { IRegisterEvent } from './IRegisterEvent';
+import {
+  AvailableCommands,
+  IAccount,
+  IAccountBalanceRequest,
+  IAccountBalanceResponse,
+  IAccountDeletionRequest,
+  IAccountDeletionResponse,
+  IAccountImportRequest,
+  IAccountImportResponse,
+  IAccountSignRequest,
+  IAccountSignResponse,
+} from '..';
+
+const MASSA_WINDOW_OBJECT = 'massaWalletProvider';
+
+type CallbackFunction = (
+  result: AllowedResponses,
+  error: Error | null,
+) => unknown;
+
+export type AllowedRequests =
+  | object
+  | IAccountBalanceRequest
+  | IAccountSignRequest
+  | IAccountImportRequest
+  | IAccountDeletionRequest;
+
+export type AllowedResponses =
+  | object
+  | IAccountBalanceResponse
+  | IAccountSignResponse
+  | IAccountImportResponse
+  | IAccountDeletionResponse
+  | IAccount[];
+
+class Connector {
+  private registeredProviders: { [key: string]: string } = {};
+  private pendingRequests: Map<string, CallbackFunction>;
+
+  public constructor() {
+    this.pendingRequests = new Map<string, CallbackFunction>();
+    this.register();
+
+    // start listening to messages from content script
+    document
+      .getElementById(MASSA_WINDOW_OBJECT)
+      .addEventListener(
+        'message',
+        this.handleResponseFromContentScript.bind(this),
+      );
+  }
+
+  private register() {
+    // global event target to use for all wallet provider
+    if (!document.getElementById(MASSA_WINDOW_OBJECT)) {
+      const inv = document.createElement('p');
+      inv.id = MASSA_WINDOW_OBJECT;
+      inv.setAttribute('style', 'display:none');
+      document.body.appendChild(inv);
+    }
+
+    // add an invisible HTML element and set a listener to it like the following
+    // hook up register handler
+    document
+      .getElementById(MASSA_WINDOW_OBJECT)
+      .addEventListener('register', (evt: CustomEvent) => {
+        const payload: IRegisterEvent = evt.detail;
+        const providerEventTargetName = `${MASSA_WINDOW_OBJECT}_${payload.providerName}`;
```

Comment:
something diverge from the standard here https://github.com/massalabs/massa-standards/blob/main/wallet/dapps-communication.md#specification

it should be

```ts
`${MASSA_WINDOW_OBJECT}-${payload.providerName}`
```